### PR TITLE
XGBoost: Add multiple notebooks for abalone data

### PR DESCRIPTION
**Pull request by iyerr3**
*Description of changes:*

This commit makes the following changes:

1. Updates the default xgboost_abalone notebook to use the latest
container
2. Adds a new notebook to describe Script mode with multiple
distributed hosts.
3. Adds a new notebook to show XGBoost training using Spot instances
4. Adds a new notebook to show usage of Parquet dataset as input to
XGBoost training

Co-authored-by: Eric Kim <dohyeong@amazon.com>
Co-authored-by: Edward Kim <kimedw@amazon.com>
Co-authored-by: Patrick Lin <patlin@amazon.com>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    "Both functions take the checkpoint directory as input, which in the below example is set to `/opt/ml/checkpoints`. \n",
+    "The primary arguments that change for the `xgb.train` call are \n",
+    "\n",
+    "1. `xgb_model`: This refers to the previous checkpoint (saved from a previously run partial job) obtained by `load_checkpoint`. This would be `None` if no previous checkpoint is available. \n",
+    "2. `callbacks`: This contains a function that performs the checkpointing\n",
+    "\n",
+    "Updated script looks like the following. \n",
+    "\n",
+    "---------\n",
+    "```\n",
```
Comment (edwardjkim) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb]: We've previously discussed making an `add_checkpointing()` function for script mode, and I just realized the change has been sitting in my branch and I didn't actually make a PR for this. Not sure why I completely forgot about this, my apologies.

PR here: https://github.com/aws/sagemaker-xgboost-container/pull/47

The above PR would change this code block to look like as follows:
```python
from sagemaker_xgboost_container.callback import add_checkpointing
CHECKPOINTS_DIR = '/opt/ml/checkpoints'   # default location for Checkpoints
callbacks = []
prev_checkpoint, n_remaining_iterations = add_checkpointing(
    callbacks, CHECKPOINTS_DIR, num_boost_round)
    params,
    dtrain, 
    num_boost_round=n_remaining_iterations,  
    xgb_model=prev_checkpoint,  
    callbacks=callbacks  
    )  
```

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sagemaker.amazon.amazon_estimator import get_image_uri\n",
+    "container = get_image_uri(region, 'xgboost', '0.90-1')"
```
Comment (edwardjkim) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb]: Should we change this to `0.90-2`?

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+    "urllib.request.urlretrieve(\"https://www.csie.ntu.edu.tw/~cjlin/libsvmtools/datasets/regression/abalone\", FILE_DATA)\n",
+    "sagemaker.Session().upload_data(FILE_DATA, bucket=bucket, key_prefix=prefix+'/train')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Obtaining the latest XGBoost container\n",
+    "We obtain the new container by specifying the framework version (0.90-1). This version specifies the upstream XGBoost framework version (0.90) and an additional SageMaker version (1). If you have an existing XGBoost workflow based on the previous (0.72) container, this would be the only change necessary to get the same workflow working with the new container."
```
Comment (edwardjkim) [inline introduction_to_amazon_algorithms/xgboost_abalone/xgboost_managed_spot_training.ipynb]: Should we change this to `0.90-2`?

File: introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    "status = client.describe_training_job(TrainingJobName=job_name)['TrainingJobStatus']\n",
+    "print(status)\n",
+    "while status !='Completed' and status!='Failed':\n",
+    "    time.sleep(60)\n",
+    "    status = client.describe_training_job(TrainingJobName=job_name)['TrainingJobStatus']\n",
+    "    print(status)"
+   ]
+  },
+  {
+   "cell_type": "code",
```
[the remark this answers is not in the document]
Comment (aws-patlin) [reply introduction_to_amazon_algorithms/xgboost_abalone/xgboost_parquet_input_training.ipynb]: Looks like this issue is resolved, and the metrics appear now.

**Pull request merged**